packages/common/src/ui/components/popover/BasePopover/usePopover.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { Dispatch, FC, useState, useRef, MutableRefObject } from 'react';
+import { useDebounceCallback } from '../../../../hooks/useDebounce';
+import { BasePopover } from './BasePopover';
+
+type VirtualElement = { getBoundingClientRect: () => DOMRect };
+
+type AnchorElementSetter = React.Dispatch<
+  React.SetStateAction<VirtualElement | null>
+>;
+
+type AnchorElementSetterRef = MutableRefObject<AnchorElementSetter | null>;
+type IsOpenSetterRef = MutableRefObject<Dispatch<
+  React.SetStateAction<boolean>
+> | null>;
+
+interface UsePopoverControl {
+  show: React.MouseEventHandler<HTMLDivElement | HTMLButtonElement>;
+  hide: () => void;
+  Popover: FC;
+}
+
+export const usePopover = (): UsePopoverControl => {
```

Comment:
I think this ends up being nice to use, though the implementation I can see does become a little confusing

Reply:
uh huh. usage is very nice though.. so implementation doesn't really affect anyone